packages/core/src/machines/actors/auth/signOut.ts

```diff
@@ -0,0 +1,27 @@
+import { createMachine } from 'xstate';
+
+import { AuthEvent, SignOutContext } from '../../../types';
+import { Auth } from 'aws-amplify';
+
+export const signOutActor = createMachine<SignOutContext, AuthEvent>(
+  {
+    initial: 'pending',
+    id: 'signOutActor',
+    states: {
+      pending: {
+        invoke: {
+          src: 'signOut',
+          onDone: 'resolved',
+          onError: 'rejected',
+        },
+      },
+      resolved: { type: 'final' },
+      rejected: { type: 'final' },
```

Comment:
Not sure why `Auth.signUp` would fail... but in that case it'd transition to `#authMachine.idle` to check current user anyway and transition to appropriate state.